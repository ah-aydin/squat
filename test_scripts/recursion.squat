func fib(int depth) int {
    if (depth == 0) {
        return 0;
    }
    if (depth == 1) {
        return 1;
    }
    var value = fib(depth - 1) + fib(depth - 2);
    return fib(depth - 1) + fib(depth - 2);
}

func main() {
    var fib_location = 10;
    print("Fibonaci number at location ");
    print(fib_location);
    print(" is ");
    print(fib(fib_location));
}
