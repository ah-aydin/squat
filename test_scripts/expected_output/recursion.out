Compiling and running file: test_scripts/recursion.squat
Fibonaci number at location 10 is 55